addons/udes_stock/views/stock_picking.xml

```diff
@@ -101,6 +101,15 @@
                 <attribute name="invisible">1</attribute>
             </xpath>
 
+            <!-- Remove Return button for all -->
+            <xpath expr="//button[@name='%(stock.act_stock_return_picking)d']" position="replace"/>
+
+            <!-- Rename to avoid confusion between discard and cancel -->
+            <xpath expr="//button[@name='action_cancel']" position="attributes">
+                <attribute name="string">Cancel Pick</attribute>
+                <attribute name="attrs">{'invisible': [('state', 'not in', ('assigned', 'confirmed', 'partially_available', 'waiting'))]}</attribute>
```

Comment:
Everyone is tripping on this double negation. It's actually invisible when the expression on the right ("state","not in", ...) is True. So it's the opposite. It's invisible when the state is not in one of those.

Reply:
right, so surely we can change it to "in" instead and just put "draft" ? As it stands in your code I think you would also have to add "done" and "cancel" otherwise the buttons will appear in those states too.

Reply:
Yes, we can switch the logic around and make it more readable if you are suggesting that.